DigitalOffice/Services/TimeManagementService/Validators/CreateWorkTimeRequestValidator.cs

```diff
@@ -0,0 +1,87 @@
+﻿using FluentValidation;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using LT.DigitalOffice.TimeManagementService.Models;
+using LT.DigitalOffice.TimeManagementService.Repositories.Filters;
+using LT.DigitalOffice.TimeManagementService.Repositories.Interfaces;
+using System.Globalization;
+
+namespace LT.DigitalOffice.TimeManagementService.Validators
+{
+    public class CreateWorkTimeRequestValidator : AbstractValidator<CreateWorkTimeRequest>
+    {
+        private IWorkTimeRepository repository;
+
+        /// <summary>
+        /// How many days ago can WorkTime be added.
+        /// </summary>
+        public const int fromDay = 3;
+        /// <summary>
+        /// How many days ahead can WorkTime be added.
+        /// </summary>
+        public const int toDay = 2;
+        /// <summary>
+        /// Limit on working hours in a row.
+        /// </summary>
+        public readonly static TimeSpan workingLimit = new TimeSpan(24, 0, 0); // 24 hours.
+
+        private readonly DateTime fromDateTime = DateTime.Now.AddDays(-fromDay);
+        private readonly DateTime toDateTime = DateTime.Now.AddDays(toDay);
+        private readonly CultureInfo culture = CultureInfo.CreateSpecificCulture("en-GB");
+
+        public CreateWorkTimeRequestValidator([FromServices] IWorkTimeRepository repository)
+        {
+            this.repository = repository;
+
+            RuleFor(wt => wt.WorkerUserId)
+                    .NotEmpty();
+
+            RuleFor(wt => wt.StartTime)
+                .NotEqual(new DateTime())
+                .Must(st => st > fromDateTime)
+                .WithMessage(date => $"WorkTime had to be filled no later than {fromDateTime.ToString(culture)}.")
+                .Must(st => st < toDateTime)
+                .WithMessage(date => $"WorkTime cannot be filled until {toDateTime.ToString(culture)}.");
+
+            RuleFor(wt => wt.EndTime)
+                .NotEqual(new DateTime());
+
+            RuleFor(wt => wt.ProjectId)
+                .NotEmpty();
+
+            RuleFor(wt => wt.Title)
+                .NotEmpty();
+
+            RuleFor(wt => wt)
+                .Must(wt => wt.StartTime < wt.EndTime)
+                .WithMessage("You cannot indicate that you worked zero hours or a negative amount.")
+                .Must(wt => wt.StartTime - wt.EndTime <= workingLimit)
+                .WithMessage(time => string.Format(
+                    "You cannot indicate that you worked more than {0} hours and {1} minutes.",
+                    workingLimit.Hours, workingLimit.Minutes))
```

Comment:
I think this statement is wrong. 
I tested it in a custom console app, and StartTime - EndTime is always negative and so is always less than workingLimit. For that, we should write "EndTime - StartTime"

But!
We should also add check that their negative diff ("StartTime - EndTime") is not less than -24 hours. That means that the user cannot post time 2 days ago more than 24 hours as well.

And, I can't figure this out yet, we should check that the time can be added only from 8 AM to 6PM, for example. Or we can set time like ("I worked from 00:00 to 04:00")?
![image](https://user-images.githubusercontent.com/61093487/89460870-afde1700-d773-11ea-9077-65f6891e065c.png)

Reply:
Yes, there is a mistake, it should be the other way around

Reply:
Above, screenshot of your diff check.
Below: screenshot of should-be diff check. I think this illustrates my point

![image](https://user-images.githubusercontent.com/61093487/89461602-dcdef980-d774-11ea-9dcc-4308d984917f.png)

Reply:
Fixed. Added tests.